google-cloud-logging/src/main/java/com/google/cloud/logging/Context.java

```diff
@@ -18,11 +18,17 @@
 
 import com.google.cloud.logging.HttpRequest.RequestMethod;
 import com.google.common.base.MoreObjects;
-import com.google.common.base.Strings;
 import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 /** Class to hold context attributes including information about {@see HttpRequest} and tracing. */
 public class Context {
+  // validate W3C trace context value on load according to the existing version format.
+  // see https://www.w3.org/TR/trace-context/#traceparent-header-field-values for details.
+  private static final Pattern W3C_TRACE_CONTEXT_FORMAT =
+      Pattern.compile(
+          "^00-(?!00000000000000000000000000000000)[0-9abcdef]{32}-(?!000000000000)[0-9abcdef]{12}-[0-9abcdef]{2}$");
```

Comment:
I think the span field is supposed to be 16 characters (see [example here](https://www.w3.org/TR/trace-context/#relationship-between-the-headers))

FYI, the nodejs library uses [slightly different regex](https://github.com/googleapis/nodejs-logging/blob/005cced4f8c1fe5f90496537e4731c2272946cb7/src/context.ts#L220), which I [copied in Python](https://github.com/googleapis/python-logging/pull/450). There may be value in trying to use the same expressions across libraries

Reply:
And you would be absolutely correct. Late hours are never good for commits :smile: